spynnaker/pyNN/models/neural_projections/connectors/from_list_connector.py

```diff
@@ -105,22 +142,22 @@ def get_n_connections_to_post_vertex_maximum(self):
         return numpy.max(numpy.bincount(self._conn_list["target"]))
 
     @overrides(AbstractConnector.get_weight_mean)
-    def get_weight_mean(self):
+    def get_weight_mean(self, weights):
         return numpy.mean(numpy.abs(self._conn_list["weight"]))
 
     @overrides(AbstractConnector.get_weight_maximum)
-    def get_weight_maximum(self):
+    def get_weight_maximum(self, weights):
         # pylint: disable=too-many-arguments
         return numpy.amax(numpy.abs(self._conn_list["weight"]))
 
     @overrides(AbstractConnector.get_weight_variance)
-    def get_weight_variance(self):
+    def get_weight_variance(self, weights):
         # pylint: disable=too-many-arguments
         return numpy.var(numpy.abs(self._conn_list["weight"]))
 
     @overrides(AbstractConnector.create_synaptic_block)
     def create_synaptic_block(
-            self, pre_slices, pre_slice_index, post_slices,
+            self, weights, delays, pre_slices, pre_slice_index, post_slices,
```

Comment:
Final place where the check is required, this time for both weights and delays.  This will have to generate the weights and delays for the block, though this might just use the internal lists.